cmd/query/app/flags_test.go

```diff
@@ -60,6 +61,56 @@ func TestQueryBuilderFlags(t *testing.T) {
 	assert.Equal(t, 10*time.Second, qOpts.MaxClockSkewAdjust)
 }
 
+func TestQueryBuilderFlagsSeparatePorts(t *testing.T) {
+	v, command := config.Viperize(AddFlags)
+	command.ParseFlags([]string{
+		"--query.static-files=/dev/null",
+		"--query.ui-config=some.json",
+		"--query.base-path=/jaeger",
+		"--query.http-server.host-port=127.0.0.1:8080",
+		"--query.additional-headers=access-control-allow-origin:blerg",
+		"--query.additional-headers=whatever:thing",
+		"--query.max-clock-skew-adjustment=10s",
+	})
+	qOpts := new(QueryOptions).InitFromViper(v, zap.NewNop())
+	assert.Equal(t, "/dev/null", qOpts.StaticAssets)
+	assert.Equal(t, "some.json", qOpts.UIConfig)
+	assert.Equal(t, "/jaeger", qOpts.BasePath)
+	assert.Equal(t, "127.0.0.1:8080", qOpts.HTTPHostPort)
+	assert.Equal(t, ports.PortToHostPort(ports.QueryGRPC), qOpts.GRPCHostPort)
+
+	assert.Equal(t, http.Header{
+		"Access-Control-Allow-Origin": []string{"blerg"},
+		"Whatever":                    []string{"thing"},
+	}, qOpts.AdditionalHeaders)
+	assert.Equal(t, 10*time.Second, qOpts.MaxClockSkewAdjust)
+}
+
+func TestQueryBuilderFlagsSeparateNoPorts(t *testing.T) {
```

Comment:
@objectiser I think this test case would verify that the change is not a breaking change.

Reply:
Could you please leave only the flags that are relevant for the test?